assisted_deployment.sh

```diff
@@ -7,18 +7,232 @@ source common.sh
+  sourceNamespace: openshift-marketplace
+EOF
+
+  wait_for_crd "localvolumes.local.storage.openshift.io"
+
+  echo "Creating local volume and storage class..."
+  cat <<EOCR | oc apply -f -
+apiVersion: local.storage.openshift.io/v1
+kind: LocalVolume
+metadata:
+  name: assisted-service
+  namespace: openshift-local-storage
+spec:
+  logLevel: Normal
+  managementState: Managed
+  storageClassDevices:
+    - devicePaths:
+        - /dev/sdb
+        - /dev/sdc
+      storageClassName: assisted-service
+      volumeMode: Filesystem
+EOCR
+}
+
+
+function deploy_hive() {
+  echo "Installing Hive..."
+
+  cat <<EOCR | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: hive-operator
+  namespace: openshift-operators
+spec:
+  installPlanApproval: Automatic
+  name: hive-operator
+  source: community-operators
+  sourceNamespace: openshift-marketplace
+EOCR
+
+  wait_for_crd "clusterdeployments.hive.openshift.io"
+}
+
+
+function subscription_config() {
+    if [ -n "${ASSISTED_SERVICE_SERVICE_IMAGE}" ]; then
+cat <<EOF
+    - name: SERVICE_IMAGE
+      value: '$ASSISTED_SERVICE_SERVICE_IMAGE'
+EOF
+    fi
+
+    if [ -n "${ASSISTED_SERVICE_INSTALLER_IMAGE}" ]; then
+cat <<EOF
+    - name: INSTALLER_IMAGE
+      value: '$ASSISTED_SERVICE_INSTALLER_IMAGE'
+EOF
+    fi
+
+    if [ -n "${ASSISTED_SERVICE_AGENT_IMAGE}" ]; then
+cat <<EOF
+    - name: AGENT_IMAGE
+      value: '$ASSISTED_SERVICE_AGENT_IMAGE'
+EOF
+    fi
+
+    if [ -n "${ASSISTED_SERVICE_DATABASE_IMAGE}" ]; then
+cat <<EOF
+    - name: DATABASE_IMAGE
+      value: '$ASSISTED_SERVICE_DATABASE_IMAGE'
+EOF
+    fi
+
+    if [ -n "${ASSISTED_SERVICE_CONTROLLER_IMAGE}" ]; then
+cat <<EOF
+    - name: CONTROLLER_IMAGE
+      value: '$ASSISTED_SERVICE_CONTROLLER_IMAGE'
+EOF
+    fi
+
+    if [ -n "${ASSISTED_SERVICE_OPENSHIFT_VERSIONS}" ]; then
+cat <<EOF
+    - name: OPENSHIFT_VERSIONS
+      value: '$ASSISTED_SERVICE_OPENSHIFT_VERSIONS'
+EOF
+    fi
+}
+
+
+function deploy_assisted_operator() {
+  echo "Installing assisted-installer operator..."
+
+  cat <<EOF | oc apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: $ASSISTED_SERVICE_NAMESPACE
+  labels:
+    name: $ASSISTED_SERVICE_NAMESPACE
+EOF
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: assisted-service-catalog
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: $ASSISTED_SERVICE_OPERATOR_INDEX
+EOF
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+    name: assisted-installer-group
+    namespace: $ASSISTED_SERVICE_NAMESPACE
+spec:
+  targetNamespaces:
+    - $ASSISTED_SERVICE_NAMESPACE
+EOF
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: assisted-service-operator
+  namespace: $ASSISTED_SERVICE_NAMESPACE
+spec:
+  config:
+    env:
+$(subscription_config)
```

Comment:
only curious what OLM does when nothing is specified here.

Reply:
I acutally tested this. This results in `conffig: {}` and things still deploy normally 

![DeepinScreenshot_select-area_20210419101938](https://user-images.githubusercontent.com/13816/115212794-cc9b5e80-a0f8-11eb-9d50-97ac8eb90eb0.png)

![DeepinScreenshot_select-area_20210419101953](https://user-images.githubusercontent.com/13816/115212809-d0c77c00-a0f8-11eb-9eb0-b9eab36bd5fd.png)